### PR TITLE
Make LedgerTrie remove work for truncated history

### PR DESCRIPTION
**Pull request by mellery451**

### CODE REVIEW

File: src/ripple/consensus/LedgerTrie.h [hunk 1]
```diff
@@ -513,58 +529,51 @@ class LedgerTrie
     bool
     remove(Ledger const& ledger, std::uint32_t count = 1)
```
Comment (nbougalis) [inline src/ripple/consensus/LedgerTrie.h]: Is calling this function with `count == 0` allowed?

Comment (nbougalis) [inline src/ripple/consensus/LedgerTrie.h]: Is calling this function with `count == 0` allowed?

File: src/ripple/consensus/LedgerTrie.h [hunk 2]
```diff
@@ -575,14 +584,8 @@ class LedgerTrie
...
-        Node const* loc;
-        Seq diffSeq;
-        std::tie(loc, diffSeq) = find(ledger);
-
-        // Exact match
-        if (loc && diffSeq == loc->span.end() && diffSeq > ledger.seq())
-            return loc->tipSupport;
-        return 0;
+        Node const* loc = findByLedgerID(ledger);
+        return loc ? loc->tipSupport : 0;
```
Comment (nbougalis) [inline src/ripple/consensus/LedgerTrie.h]: This is more of a stylistic nit than anything, but what about:

    if (auto const* loc = findLedgerID(ledger))
        return loc->tipSupport;
    return 0;

File: src/ripple/consensus/LedgerTrie.h [hunk 3]
```diff
@@ -395,6 +395,22 @@ class LedgerTrie
         return std::make_pair(curr, pos);
     }
 
+    Node*
```
Comment (bachase) [inline src/ripple/consensus/LedgerTrie.h]: Nit: Consider adding documentation for this function. Its internal, but might be helpful nonetheless.

File: src/ripple/consensus/LedgerTrie.h (hunk 1 above)
Comment (bachase) [reply src/ripple/consensus/LedgerTrie.h]: It should work but not change anything in the trie.

**Pull request closed without merging**